pkg/storageincentives/agent.go

```diff
@@ -323,6 +336,16 @@ func (a *Agent) play(ctx context.Context) (uint8, []byte, error) {
 
 	storageRadius := a.radius.StorageRadius()
 
+	// true if frozen
+	isFrozen, err := a.redistributionStatuser.IsOverlayFrozen(ctx)
+	if err != nil {
+		a.logger.Info("error checking if stake is frozen", "err", err)
+	}
+	if isFrozen {
+		a.nodeState.SetState(frozen)
+		a.nodeState.SaveStatus()
+	}
+
```

Comment:
it would be good to have a lastPlayedRound field as well

Reply:
Done